Running groovy scripts on windows fails on compiling (-g The system cannot find the file specified)

**Describe the bug**
Trying to run the hello.groovy script on windows (both git bash & windows cmd)

C:\work\misc\groovy-scratch\src>jbang --verbose test.groovy
[jbang] [3:013] jbang version 0.109.0
[jbang] [3:278] Resolving resource ref: test.groovy
[jbang] [3:328] Resolved resource ref as: test.groovy (cached as: C:\work\misc\groovy-scratch\src\test.groovy)
[jbang] [3:559] Build required as C:\Users\SamuliSaarinen\.jbang\cache\jars\test.groovy.2a85157bad3bb835195b0c4a19ed409de05d6d16560310c04b96e43ebeaf8b34\test.jar not readable or not found.
[jbang] [3:579] Resolving artifact(s): org.codehaus.groovy:groovy:3.0.9
[jbang] [3:596] Repositories: mavencentral=https://repo1.maven.org/maven2/
[jbang] [3:833] Resolved artifact(s) from cache: org.codehaus.groovy:groovy:jar:3.0.9=C:\Users\SamuliSaarinen\.m2\repository\org\codehaus\groovy\groovy\3.0.9\groovy-3.0.9.jar
[jbang] [4:129] Building jar for test.groovy...
[jbang] [4:129] Compile: C:\Users\SamuliSaarinen\.jbang\cache\groovycs\3.0.9\groovy-3.0.9\bin\groovyc.bat -g -classpath C:\Users\SamuliSaarinen\.m2\repository\org\codehaus\groovy\groovy\3.0.9\groovy-3.0.9.jar -d C:\Users\SamuliSaarinen\.jbang\cache\jars\test.groovy.2a85157bad3bb835195b0c4a19ed409de05d6d16560310c04b96e43ebeaf8b34\classes C:\work\misc\groovy-scratch\src\test.groovy
[jbang] [4:151] Looking for JDK: 0
[jbang] [4:228] Using JDK: 17 (17.0.8+7, current, C:\Users\SamuliSaarinen\.jbang\cache\jdks\17)
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
-g: -g (The system cannot find the file specified)

1 error

[jbang] [14:421] Deleting folder C:\Users\SamuliSaarinen\.jbang\cache\jars\test.groovy.2a85157bad3bb835195b0c4a19ed409de05d6d16560310c04b96e43ebeaf8b34\classes
[jbang] [14:437] Deleting folder C:\Users\SamuliSaarinen\.jbang\cache\jars\test.groovy.2a85157bad3bb835195b0c4a19ed409de05d6d16560310c04b96e43ebeaf8b34\generated
[jbang] [14:437] [ERROR] Error during compile
dev.jbang.cli.ExitException: Error during compile
        at dev.jbang.source.buildsteps.CompileBuildStep.runCompiler(CompileBuildStep.java:145)
        at dev.jbang.source.sources.GroovySource$GroovyAppBuilder$GroovyCompileBuildStep.runCompiler(GroovySource.java:97)
        at dev.jbang.source.buildsteps.CompileBuildStep.runCompiler(CompileBuildStep.java:133)
        at dev.jbang.source.buildsteps.CompileBuildStep.compile(CompileBuildStep.java:117)
        at dev.jbang.source.buildsteps.CompileBuildStep.build(CompileBuildStep.java:56)
        at dev.jbang.source.buildsteps.CompileBuildStep.build(CompileBuildStep.java:41)
        at dev.jbang.source.AppBuilder.build(AppBuilder.java:97)
        at dev.jbang.source.AppBuilder.build(AppBuilder.java:22)
        at dev.jbang.cli.Run.doCall(Run.java:89)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:145)
        at dev.jbang.cli.BaseCommand.call(BaseCommand.java:21)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
        at dev.jbang.cli.JBang$3.handle(JBang.java:148)
        at dev.jbang.cli.JBang$3.handle(JBang.java:143)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at dev.jbang.Main.main(Main.java:14)

**Additional context**
Seems like there is a `-g` command line flag added to groovyc but I can't find it in the groovyc documentation and it seems to cause the error. Running the groovyc command without -g runs fine

e.g.
```
C:\Users\SamuliSaarinen\.jbang\cache\groovycs\3.0.9\groovy-3.0.9\bin\groovyc.bat -classpath C:\Users\SamuliSaarinen\.m2\repository\org\codehaus\groovy\groovy\3.0.9\groovy-3.0.9.jar -d C:\Users\SamuliSaarinen\.jbang\cache\jars\test.groovy.2a85157bad3bb835195b0c4a19ed409de05d6d16560310c04b96e43ebeaf8b34\classes C:\work\misc\groovy-scratch\src\test.groovy
```

--- a/src/main/java/dev/jbang/source/ProjectBuilder.java
+++ b/src/main/java/dev/jbang/source/ProjectBuilder.java
@@ -26,6 +26,7 @@ import dev.jbang.cli.ExitException;
 import dev.jbang.dependencies.*;
 import dev.jbang.source.buildsteps.JarBuildStep;
 import dev.jbang.source.resolvers.*;
+import dev.jbang.source.sources.JavaSource;
 import dev.jbang.util.JavaUtil;
 import dev.jbang.util.ModuleUtil;
 import dev.jbang.util.PropertiesValueResolver;
@@ -464,7 +465,9 @@ public class ProjectBuilder {
 		prj.setGav(src.tagReader.getGav().orElse(null));
 		prj.setMainClass(src.tagReader.getMain().orElse(null));
 		prj.setModuleName(src.tagReader.getModule().orElse(null));
-		prj.getMainSourceSet().addCompileOption("-g");
+		if (prj.getMainSource() instanceof JavaSource) {
+			prj.getMainSourceSet().addCompileOption("-g");
+		}
 		return updateProject(src, prj, resolver);
 	}
 

--- a/src/main/java/dev/jbang/source/sources/GroovySource.java
+++ b/src/main/java/dev/jbang/source/sources/GroovySource.java
@@ -23,6 +23,10 @@ public class GroovySource extends Source {
 		super(script, replaceProperties);
 	}
 
+	public GroovySource(String script, Function<String, String> replaceProperties) {
+		super(script, replaceProperties);
+	}
+
 	@Override
 	protected List<String> getCompileOptions() {
 		return tagReader.collectOptions("COMPILE_OPTIONS");

--- a/src/main/java/dev/jbang/source/sources/KotlinSource.java
+++ b/src/main/java/dev/jbang/source/sources/KotlinSource.java
@@ -19,6 +19,10 @@ public class KotlinSource extends Source {
 		super(script, replaceProperties);
 	}
 
+	public KotlinSource(String script, Function<String, String> replaceProperties) {
+		super(script, replaceProperties);
+	}
+
 	@Override
 	protected List<String> getCompileOptions() {
 		return tagReader.collectOptions("COMPILE_OPTIONS");

--- a/src/test/java/dev/jbang/source/TestSource.java
+++ b/src/test/java/dev/jbang/source/TestSource.java
@@ -20,7 +20,9 @@ import org.junit.jupiter.api.io.TempDir;
 
 import dev.jbang.BaseTest;
 import dev.jbang.net.TrustedSources;
+import dev.jbang.source.sources.GroovySource;
 import dev.jbang.source.sources.JavaSource;
+import dev.jbang.source.sources.KotlinSource;
 import dev.jbang.util.PropertiesValueResolver;
 
 public class TestSource extends BaseTest {
@@ -150,6 +152,28 @@ public class TestSource extends BaseTest {
 			"//COMPILE_OPTIONS commons-codec:commons-codec:1.15 // <.>\n" +
 			"public class test {" +
 			"}";
+
+	String groovyExample = "///usr/bin/env jbang \"$0\" \"$@\" ; exit $?\n" +
+			"//DEPS info.picocli:picocli:4.6.3\n" +
+			"\n" +
+			"//RUNTIME_OPTIONS --enable-preview \"-Dvalue='this is space'\"\n" +
+			"//JAVA_OPTIONS --enable-preview\n" +
+			"//COMPILE_OPTIONS --enable-preview\n" +
+			"//NATIVE_OPTIONS -O1\n" +
+			"\n" +
+			"println(\"Hello World\");\n";
+
+	String kotlinExample = "///usr/bin/env jbang \"$0\" \"$@\" ; exit $?\n" +
+			"//DEPS info.picocli:picocli:4.6.3\n" +
+			"\n" +
+			"//RUNTIME_OPTIONS --enable-preview \"-Dvalue='this is space'\"\n" +
+			"//JAVA_OPTIONS --enable-preview\n" +
+			"//COMPILE_OPTIONS --enable-preview\n" +
+			"//NATIVE_OPTIONS -O1\n" +
+			"\n" +
+			"public fun main() {\n" +
+			"    println(\"Hello World\");\n" +
+			"}\n";
 
 	@Test
 	void testCommentsDoesNotGetPickedUp() {
@@ -281,10 +305,31 @@ public class TestSource extends BaseTest {
 	}
 
 	@Test
-	void testExtractOptions() {
+	void testExtractJavaOptions() {
 		Source s = new JavaSource(example, null);
 
 		assertEquals(Arrays.asList("--verbose", "--enable-preview"), s.getCompileOptions());
+		assertEquals(Arrays.asList("-O1"), s.getNativeOptions());
+		assertEquals(Arrays.asList("--enable-preview", "--enable-preview", "-Dvalue='this is space'"),
+				s.getRuntimeOptions());
+
+	}
+
+	@Test
+	void testExtractGroovyOptions() {
+		Source s = new GroovySource(groovyExample, null);
+
+		assertEquals(Arrays.asList("--enable-preview"), s.getCompileOptions());
+		assertEquals(Arrays.asList("-O1"), s.getNativeOptions());
+		assertEquals(Arrays.asList("-Dgroovy.grape.enable=false", "--enable-preview", "--enable-preview",
+				"-Dvalue='this is space'"), s.getRuntimeOptions());
+	}
+
+	@Test
+	void testExtractKotlinOptions() {
+		Source s = new KotlinSource(kotlinExample, null);
+
+		assertEquals(Arrays.asList("--enable-preview"), s.getCompileOptions());
 		assertEquals(Arrays.asList("-O1"), s.getNativeOptions());
 		assertEquals(Arrays.asList("--enable-preview", "--enable-preview", "-Dvalue='this is space'"),
 				s.getRuntimeOptions());

